database/misc/mongodb_replication.py

```diff
@@ -0,0 +1,378 @@
+            timeout = timeout - 5
+            if timeout <= 0:
+                module.fail_json(msg='reached timeout while waiting for rs.reconfig(): %s' % str(e))
+            time.sleep(5)
+
+def remove_host(module, client, host_name, timeout=180):
+    while True:
+        try:
+            admin_db = client['admin']
+            local_db = client['local']
+
+            if local_db.system.replset.count() > 1:
+                module.fail_json(msg='local.system.replset has unexpected contents')
+
+            cfg = local_db.system.replset.find_one()
+            if not cfg:
+                module.fail_json(msg='no config object retrievable from local.system.replset')
+
+            cfg['version'] += 1
+
+            if len(cfg['members']) == 1:
+                module.fail_json(msg="You can't delete last member of replica set")
+            for member in cfg['members']:
+                if host_name in member['host']:
+                    cfg['members'].remove(member)
+                else:
+                    fail_msg = "couldn't find member with hostname: {0} in replica set members list".format(host_name)
+                    module.fail_json(msg=fail_msg)
+        except (OperationFailure, AutoReconnect) as e:
+            timeout = timeout - 5
+            if timeout <= 0:
+                module.fail_json(msg='reached timeout while waiting for rs.reconfig(): %s' % str(e))
+            time.sleep(5)
+
+def load_mongocnf():
+    config = ConfigParser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
+        creds = dict(
+          user=config.get('client', 'user'),
+          password=config.get('client', 'pass')
+        )
+    except (ConfigParser.NoOptionError, IOError):
+        return False
+
+    return creds
+
+def wait_for_ok_and_master(module, client, timeout = 60):
+    while True:
+        status = client.admin.command('replSetGetStatus', check=False)
+        if status['ok'] == 1 and status['myState'] == 1:
+            return
+
+        timeout = timeout - 1
+        if timeout == 0:
+            module.fail_json(msg='reached timeout while waiting for rs.status() to become ok=1')
+
+        time.sleep(1)
+
+def authenticate(client, login_user, login_password):
+    if login_user is None and login_password is None:
+        mongocnf_creds = load_mongocnf()
+        if mongocnf_creds is not False:
+            login_user = mongocnf_creds['user']
+            login_password = mongocnf_creds['password']
+        elif login_password is None and login_user is not None:
+            module.fail_json(msg='when supplying login arguments, both login_user and login_password must be provided')
+
+    if login_user is not None and login_password is not None:
+        client.admin.authenticate(login_user, login_password)
+# =========================================
+# Module execution.
+#
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
```

Comment:
Because this params(login_user, login_password) not required, because you can connect to host without authentication.